### ISSUE TITLE
Wrong overload for `str.join`

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#306]
The return type of `str.join` should be `str` here, not `LiteralString`:
```py
from typing import reveal_type

def _(pair: tuple[str, str]):
    reveal_type(", ".join(pair))  # ty: LiteralString
```

**Comment by dhruvmanila** [comment]
The first example is probably related to https://github.com/astral-sh/ty/issues/665.

**Comment by carljm** [comment]
And I suspect the second part is because `Iterable` is a protocol, and currently our protocol support only checks member existence, not member type; @AlexWaygood is working on this.

**Comment by karlicoss** [comment]
I have the following example where I'd expect type error, I assume it's the same issue? https://play.ty.dev/5294987b-6776-4cde-9844-bf45e510f59d
```python
from collections.abc import Iterator, Sequence

def gen() -> Iterator[str]:
    yield 123  # expect type error here!

def seq() -> Sequence[str]:
    return [123]  # expect type error here!


a: Iterator[int] = gen()  # expect type error here!

## the types themselves seem to be inferred correctly
reveal_type(gen)
reveal_type(seq())
```

**Comment by jelle-openai** [comment]
I'd suspect that's different. The `return [123]` is because ty isn't inferring precise types for list literals; there are open issues about that. The `yield 123` likely means that ty isn't type-checking yields in generators yet. I don't see an open issue about that but [experimenting in the playground](https://play.ty.dev/ffa8537a-9610-4958-91cf-fdf9f1b429ed) gives some TODO types.
